### PR TITLE
making slicing more pythonic

### PR DESCRIPTION
**Pull request by abit2**
This way it makes cube_sequence.index_as_cube[0:5] possible which is much better than cube_sequence.index_as_cube[slice(0,5)].
@DanRyanIrish 
could not come up with a better name open to renaming the helper class :)

### CODE REVIEW

File: sunpycube/cube/datacube.py [hunk 1]
```diff
@@ -639,3 +640,13 @@ def plot_wavelength_slice(self, offset, **kwargs):
             offset, cumul_cube_lengths)
         plot = self[sequence_index].plot_wavelength_slice(cube_index, **kwargs)
         return plot
+
+
+class Slice_helper(object):
+    """Helper class to make slicing in index_as_cube more pythonic"""
```
Comment (DanRyanIrish) [inline sunpycube/cube/datacube.py]: Can you be a little more explicit in you docstring?  Imagine documenting it so one of the IRIS team members, who haven't been following the details of our development, can get a idea of what the purpose of this helper is and how it works.

File: sunpycube/cube/datacube.py [hunk 2]
```diff
@@ -582,11 +582,12 @@ def _new_instance(cls, data_list, meta=None, common_axis=None):
         """
         return cls(data_list, meta=meta, common_axis=common_axis)
 
-    def index_as_cube(self, item):
+    @property
+    def index_as_cube(self):
```
Comment (DanRyanIrish) [inline sunpycube/cube/datacube.py:586]: Could you include an example in this docstring of how this function should be used?

File: sunpycube/cube/datacube.py [hunk 3]
```diff
@@ -639,3 +640,13 @@ def plot_wavelength_slice(self, offset, **kwargs):
             offset, cumul_cube_lengths)
         plot = self[sequence_index].plot_wavelength_slice(cube_index, **kwargs)
         return plot
+
+
+class Slice_helper(object):
+    """Helper class to make slicing in index_as_cube more pythonic"""
+
+    def __init__(self, seq):
+        self.seq = seq
```
Comment (DanRyanIrish) [inline sunpycube/cube/datacube.py]: Can you also explain in a comment or docstring what ```seq``` is?  Just a line should do.

File: sunpycube/cube/datacube.py [hunk 4]
```diff
@@ -639,3 +640,13 @@ def plot_wavelength_slice(self, offset, **kwargs):
...
+
+    def __getitem__(self, item):
```
Comment (DanRyanIrish) [inline sunpycube/cube/datacube.py:673]: Either here or in one of the docstrings above can you list the different allowed types that item can take?

**Review by DanRyanIrish — CHANGES_REQUESTED** [review]

**Comment by DanRyanIrish** [comment]
Hi @abit2.  It's great you've already made headway with this.  Just left some comments to improve the documentation so someone else (or our future selves!) can easily see what is going on here.  Great work so far!

**Comment by DanRyanIrish** [comment]
Regarding the helper class name, yeah it's tough to come up with a good one.  ```IndexAsCubeSlicer```?  Any other thoughts?  @Cadair?  This class should probably be "hidden" as well.  No need for typical users to see it.  So if the name is as above, it should actually be ```_IndexAsCubeSlicer```

**Comment by abit2** [comment]
@DanRyanIrish updated :)

**Review by DanRyanIrish — APPROVED** [review]

**Pull request merged**